### ISSUE TITLE
Attach gpg keys

### ISSUE THREAD

**Issue by varac** [astroidmail/astroid#600]
Coming from alot, I'm missing the functionality to attache gpg keys.
There are [contrib hooks in the alot wiki](https://github.com/pazz/alot/wiki/Contrib-Hooks#attach-gpg-keys) which allows attaching either one of:

- [ ] the senders default gpg key
- [ ] all recipient keys
- [ ] an arbitriray gpg key

It would be great if all of them would be included in astroid.

**Comment by gauteh** [comment]
I agree, we miss an equivalent to process message for messages being sent
(think filter Forward headers, implement autocrypt as a plugin, etc).
Keyboard hooks that can work on the edited message is also a good idea,
though direct access to the attachment vector might be tricky. It is
probably useful enough to implement as a special case, otherwise the
necessary plugin interface could be exposed.
